developers/weaviate/client-libraries/typescript/v2_v3_migration.md

```diff
@@ -129,132 +101,308 @@ Once it has been instantiated, you will notice that the client API is different
 
-  <FilteredTextBlock
-    text={QuickStartCode}
-    startMarker="# NearTextExample"
-    endMarker="# END NearTextExample"
-    language="py"
-  />
+```ts
+const collection = client.collections.get<Article>('Article');
+
+await collection.insert({ // compiler error since 'body' field is missing in '.insert'
+  title: 'TS is awesome!',
+  wordcount: 9001
+})
+```
 
 </TabItem>
 </Tabs>
 
-In both of these examples, you can see how the helper classes and methods abstract away the need for manual JSONs or strings.
+As in the example above, we can extend the use of Generics to create type-safe operations around collection creation, querying and deletion to mention a few. 
 
 ### Interaction with collections
 
 Interacting with the `client` object for CRUD and search operations have been replaced with the use of collection objects.
 
 This conveniently removes the need to specify the collection for each operation, and reduces potential for errors.
 
-import ManageDataCode from '!!raw-loader!/_includes/code/howto/manage-data.read.py';
-import ManageDataCodeV3 from '!!raw-loader!/_includes/code/howto/manage-data.read-v3.py';
-
 <Tabs groupId="languages">
-  <TabItem value="py" label="Python (v4)">
-    <FilteredTextBlock
-      text={ManageDataCode}
-      startMarker="# ReadObject START"
-      endMarker="# ReadObject END"
-      language="py"
-    />
-  </TabItem>
-
-  <TabItem value="py3" label="Python (v3)">
-    <FilteredTextBlock
-      text={ManageDataCodeV3}
-      startMarker="# ReadObject START"
-      endMarker="# ReadObject END"
-      language="py"
-    />
-  </TabItem>
+<TabItem value="jsv3" label="JS/TS (v3)">
+
+```ts
+const myCollection = client.collections.get('JeopardyQuestion');
+
+const result = await myCollection.query.fetchObjects({
+  returnProperties: ['question'],
+})
+
+console.log(JSON.stringify(result.objects, null, 2));
+```
+
+</TabItem>
+<TabItem value="jsv2" label="JS/TS (v2)">
+
+```ts
+result = await client
+  .graphql
+  .get()
+  .withClassName('JeopardyQuestion')
+  .withFields('question')
+  .do();
+
+console.log(JSON.stringify(result, null, 2));
+```
+
+</TabItem>
 </Tabs>
 
 Note here that the collection object can be re-used throughout the codebase.
 
-### Collection creation from JSON
 
-You can still create a collection from a JSON definition. This may be a useful way to migrate your existing data, for example. You could [fetch an existing definition](../../manage-data/collections.mdx#read-a-single-collection-definition) and then use it to create a new collection.
+### Removal of Builder Pattern
 
-<FilteredTextBlock
-  text={PythonCode}
-  startMarker="# START CreateCollectionFromJSON"
-  endMarker="# END CreateCollectionFromJSON"
-  language="py"
-/>
+The builder patterns for constructing queries as been removed, as they could be confusing and potentially lead to invalid queries.
 
-### Removal of builder patterns
+In `v3`, queries are constructed using specific methods and its parameters.
```

Comment:
Consider:

In `v3`, use specific methods and their built-in parameters to construct queries.